api/http/handler/endpoints/endpoint_create.go

```diff
@@ -66,6 +67,14 @@ func (payload *endpointCreatePayload) Validate(r *http.Request) error {
 	}
 	payload.EndpointCreationType = endpointCreationEnum(endpointCreationType)
 
+	payload.ContainerEngine, err = request.RetrieveMultiPartFormValue(r, "ContainerEngine", true)
```

Comment:
I think it's better to always have error handling before any kind of default logic.

```suggestion
	payload.ContainerEngine, err = request.RetrieveMultiPartFormValue(r, "ContainerEngine", true)
	if err != nil || (payload.ContainerEngine != "" && payload.ContainerEngine != portainer.ContainerEngineDocker && payload.ContainerEngine != portainer.ContainerEnginePodman) {
		return errors.New("invalid container engine value. Value must be one of: 'docker' or 'podman'")
	}
	if payload.ContainerEngine == "" {
		payload.ContainerEngine = portainer.ContainerEngineDocker
	}
```

Reply:
updated